### DISCUSSION TITLE
Rows updated on the grid does not always update

### DISCUSSION THREAD

**Discussion by arlem03** [dbeaver/dbeaver#35826] [Q&A] [answered]
### Description

I am having an unexpected issue. When updating rows using the query grid result, and then save, I often found that the database didn't update. Yesterday, I updated about 100 rows--one at the time--and found that about 40 never updated even when it shows at the bottom of the grid that the rows were updated. If I open another query and pull the same records, I find them not updated. I am not sure what this is, or if this has to do with this specific version. I will try an earlier version and see if I get the same result.

### DBeaver Version

24.2

### Operating System

Linux

### Database and driver

MariaDB

### Steps to reproduce

1. Connect to the database.
2. Do a query such as "SELECT * FROM stills.db.stills WHERE sharedid = 6644;
3.  I get a grid result showing two rows.
4.  I select the filename cell under the filename column (background for the row becomes orange), and then change the data in the cell. 
5.  I press Save or CTRL-v. The cell background color return to normal, and the botton indicate shows the 2 rows have been updated.
6. I open a new query and select the same rows. They show are not updated. I try several times--same result. I restart DBeaver and get the same result.

### Additional context

The updates are done remotely using a linux workstation. The database is also on a linux server.

**Comment by E1izabeth** [comment, marked as answer]
DBeaver executes those updates in Transactions. The transaction state is shown in the Main toolbar at the top, and the counter shows the count of queries in the current transaction.
![image](https://github.com/user-attachments/assets/3576eb4e-f480-43f8-ab57-6583d6e88018)
To apply changes, please press the Commit button.

If you want your changes to be applied immediately, please uncheck the `Smart commit mode` option here
![image](https://github.com/user-attachments/assets/4d4be24b-dc77-40ec-8a4e-9c2f5789a98e)
or in `Connection configuration->Connection settings->Transactions`
![image](https://github.com/user-attachments/assets/44b1d707-e227-4170-8fc2-58a5f5ef7861)

**Comment by arlem03** [comment]
Thank you Elizabeth, 

Right! Somehow I thought the commit was included in the saving option. I did find, in the documentation, that I could auto-commit by default. So I set it up and now understand why the development connection type auto-commit and why the production doesn't auto-commit. So when I changed from Dev to Prod, I didn't notice that the auto-commit was not enabled on the Prod connection type. This is now resolved. Thank you for your directions. 

Robert

**Comment by arlem03** [comment]
Thank you Elizabeth, 

Right! Somehow I thought the commit was included in the saving option. 

Here is a question that might sound stupid to you. Can I do a commit after several queries and changes, or do I have to do a commit after every query and save? 

Robert

**Comment by E1izabeth** [comment]
Yes, you can do Commit after several changes
